src/server/editorServices.ts

```diff
@@ -2083,9 +2102,34 @@ namespace ts.server {
 
             this.printProjects();
 
+            this.telemetryOnOpenFile(fileName, info);
             return { configFileName, configFileErrors };
         }
 
+        private telemetryOnOpenFile(fileName: NormalizedPath, info: ScriptInfo): void {
+            Debug.assert(info.containingProjects.length !== 0);
+            if (
+                !this.eventHandler ||
+                // Don't send for a project with 'checkJs' enabled globally.
+                info.getDefaultProject().getCompilationSettings().checkJs ||
+                // We are only sending stats for JS files
+                !info.isJavaScript() ||
+                // If we've already seen the file, no need to send a new event.
+                this.allJsFilesForOpenFilesTelemetry.has(fileName)) {

```

Comment:
Should use filePath as a key instead of fileName since the caseInsensitive system: "filename" === "fileName"

Reply:
use info.path instead.